File: Code/Java/trimorphic_number.java
/*
 * A trimorphic number is whose cube ends with the number itself.
 * Example:
 *      49*49*49 = 117649
 *      which ends with 49
 * This code checks whehter the input number is trimorphic or not.
 */
import java.util.*;
class trimorphic_number
{
    void main()
    {
        //driver function
        
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter number: ");
        int number = scanner.nextInt();
        scanner.nextLine();

        if(is_trimorphic(number) == true)
        {
            System.out.println(number+" is a Trimorphic number!");
        }
        else
        {
            System.out.println(number+" is not a Trimorphic number!");
        }
    }
    
    boolean is_trimorphic(int number)
    {
        //this function checks whether number is trimorphic or not
        
        int cube = number*number*number;
        
        if((cube % Math.pow(10,no_of_digits(number))) == number)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    
    int no_of_digits(int number)
    {
        //this function counts the number of digits
        
        int count = 0;
        
        while(number!=0)
        {
            count++;
            number/=10;
        }
        
        return count;
    }
}

/*
 * Test Cases-
 * 
 * 1.
 * Enter number: 
 * 24
 * 24 is a Trimorphic number!
 * 
 * 2.
 * Enter number: 
 * 21
 * 21 is not a Trimorphic number!
 *
 * Time Complexity: O(n)
 * where n is the number of digits in the number
 * Space Complexity: O(1)
 */

File: Code/Python/TrimorphicNumber.py
''' 
-  In this program we have to tell whether a given number
   is trimorphic or not.
-  A trimorphic number is a number whose cube's last digit 
   is equal to the last digit of the number itself.
-   For Ex:  
           If number=10
               cube=1000
            Since last digit to 1000 and 10 both are 0. 
            Therefore, 10 is a Trimorphic number.
'''

num=int(input())
flag=0
cube_power=num*num*num
while num!=0:
    if num%10!=cube_power%10:
        flag=1
        break
    num//=10
    cube_power//=10
if flag==0:
    print("The given number is Trimorphic")
else:
   print("The given is not number is Trimorphic")

'''    TEST CASES
    
    1)Input: 
            Enter the Number
            24
    2)Output:
            The given number is Trimorphic
    
     1)Input: 
            Enter the Number
            2
    2)Output:
            The given number is not Trimorphic

    Time Complexity:O(1)
    Space Complexity:O(1)
'''